.github/workflows/shortIntegrationTests.yml

```diff
@@ -0,0 +1,86 @@
+name: Short integration tests
+
+# Run the integration tests on one configuration (of the nightly matrix) on each
+# push to main. Should catch 95% of integration test breakages. Useful for
+# identifying the particular offending commit and emailing the commit author.
+
+# Note: currently the integration tests don't take that long to run (<1 min).
+# If, in the future, they take significantly longer (>~1 hour), we should
+# reconsider this.
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  # Build CIRCT and run its tests using a Docker container with all the
+  # integration testing prerequisite installed.
+  build-circt:
+    name: Build and Test
+    runs-on: ubuntu-18.04
+    container:
+      image: ghcr.io/circt/images/circt-integration-test:v6-beta3
+    strategy:
+      # Keep the 'matrix' strategy with one data point to make it obvious that
+      # this is one point in the overall matrix.
+      matrix:
+        build-assert: [ON]
+        build-shared: [OFF]
```

Comment:
I'd recommend ON for this, as some linkage [issues](https://github.com/llvm/circt/commit/8dca4c38ecef4fb5251ef7335da084bacb94b56a) only crop up with that flag on.

Reply:
Doesn't `ON` disable the Python bindings?

Reply:
I don't think so, I've been using BUILD_SHARED_LIBS=ON for local development. We default to [OFF for unified builds](https://github.com/llvm/circt/blob/main/CMakeLists.txt#L79), but there shouldn't be any issue in overriding it.

Reply:
There's also a little workaround (separate, and can be removed now), which is a vestige of me getting BUILD_SHARED_LIBS working for Python: https://github.com/llvm/circt/blob/main/lib/Bindings/Python/CMakeLists.txt#L16